### PR TITLE
notify when office hours goes live

### LINKED ISSUE #278: notify when our favourite office hours goes live
DuckBot (or discord, if built in) should notify us of when office hours starts, so we can get our questions answered.

https://www.twitch.tv/conlabx  :brother:

### PR DESCRIPTION
**Pull request by twentylemon**
##### Summary 

Fixes #278

The twitch api is a bit of a pain, so I opted to just poll the stream page. Nice answer on [stackoverflow](https://stackoverflow.com/a/67969583) is the solution I used. For fun (since AWS $$$), I also reduce the polling once we find office hours have started.

##### Checklist

* [x] this is a source code change
  * [x] run `isort . && black .` in the repository root
  * [x] run `pytest` in the repository root
  * [x] link to issue being fixed using a [_fixes keyword_](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), if such an issue exists
  * [x] update the wiki documentation if necessary
* [ ] or, this is **not** a source code change

### CODE REVIEW

File: duckbot/cogs/games/office_hours.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    def cog_unload(self):
+        self.check_if_streaming_loop.cancel()
+
+    @tasks.loop(minutes=15.0)
+    async def check_if_streaming_loop(self):
+        await self.check_if_streaming()
+
+    async def check_if_streaming(self):
+        page = requests.get("https://www.twitch.tv/conlabx").content.decode("utf-8")
+        streaming = "isLiveBroadcast" in page
```
Comment (jameshughes89) [inline duckbot/cogs/games/office_hours.py:22]: amazingly easy. Nice

File: duckbot/cogs/games/office_hours.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if streaming != self.streaming:
+            self.streaming = streaming
+            if streaming:
+                channel = get(self.bot.get_all_channels(), guild__name="Friends Chat", name="general", type=ChannelType.text)
+                await channel.send("Office Hours have started!\nhttps://www.twitch.tv/conlabx")
+                self.check_if_streaming_loop.change_interval(hours=12.0)  # avoid polling for a while
+            else:
+                self.check_if_streaming_loop.change_interval(minutes=15.0)  # stream stopped, restart polling
```
Comment (jameshughes89) [inline duckbot/cogs/games/office_hours.py]: Could probably get fancy and make it only check ceertain times of the day. Evenings mostly.

File: duckbot/cogs/games/office_hours.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        await self.check_if_streaming()
+
+    async def check_if_streaming(self):
+        page = requests.get("https://www.twitch.tv/conlabx").content.decode("utf-8")
+        streaming = "isLiveBroadcast" in page
+        if streaming != self.streaming:
+            self.streaming = streaming
+            if streaming:
+                channel = get(self.bot.get_all_channels(), guild__name="Friends Chat", name="general", type=ChannelType.text)
+                await channel.send("Office Hours have started!\nhttps://www.twitch.tv/conlabx")
```
Comment (jameshughes89) [inline duckbot/cogs/games/office_hours.py]: Perhaps `"\"Office Hours\" have started!\nhttps://www.twitch.tv/conlabx"`

File: tests/cogs/games/office_hours_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+async def test_check_if_streaming_stream_ongoing(get, bot, general_channel):
+    clazz = OfficeHours(bot)
+    clazz.streaming = True
+    await clazz.check_if_streaming()
+    general_channel.send.assert_not_called()
+    assert clazz.streaming
+
+
+@pytest.mark.asyncio
+@mock.patch("requests.get", return_value=response("bruh"))
```
Comment (jameshughes89) [inline tests/cogs/games/office_hours_test.py:55]: bruuuuuuuuuh

**Review by jameshughes89 — APPROVED** [review]
dumplings

File: duckbot/cogs/games/office_hours.py (hunk 2 above)
Comment (twentylemon) [reply duckbot/cogs/games/office_hours.py]: I'll leave this for now. If the data transfer amount on aws is a lot, we can revisit to reduce the total number of calls.

File: duckbot/cogs/games/office_hours.py (hunk 3 above)
Comment (twentylemon) [reply duckbot/cogs/games/office_hours.py]: Sure, done.

**Pull request merged**